lib/tasks/valid_test_data_generator.rb

```diff
@@ -151,4 +153,80 @@ def random_or_nil_trn
       TRNGenerator.next
     end
   end
+
+  class NPQLeadProviderPopulater
+    class << self
+      def call(name:, total_schools: 10, participants_per_school: 10)
+        new(name: name).call(total_schools: total_schools, participants_per_school: participants_per_school)
+      end
+    end
+
+    def call(total_schools: 10, participants_per_school: 100)
+      generate_new_schools(count: total_schools, participants_per_school: participants_per_school)
+    end
+
+  private
+
+    attr_reader :lead_provider
+
+    def initialize(name:)
+      @lead_provider = ::NPQLeadProvider.find_or_create_by!(name: name)
+    end
+
+    def generate_new_schools(count:, participants_per_school:)
+      count.times { create_fip_school_with_cohort(urn: SchoolURNGenerator.next, participants_per_school: participants_per_school) }
+    end
+
+    def find_or_create_participants(school:, number_of_participants:)
+      generate_new_participants(school: school, count: number_of_participants)
+    end
+
+    def generate_new_participants(school:, count:)
+      while count.positive?
```

Comment:
`count.times do ` ?

Reply:
if you use this suggestion make sure to add the argument as line 187 will need it, probably don't call it `count` again though

Reply:
Yes. It was like this for ECF because of the variable number of ECT/Mentor and isn't required here.